client/allocrunner/consul_hook.go

```diff
@@ -175,8 +169,19 @@ func (h *consulHook) prepareConsulTokensForServices(services []*structs.Service,
 			return fmt.Errorf("no such consul cluster: %s", clusterName)
 		}
 
-		req, err := h.prepareConsulClientReq(
-			*service.IdentityHandle(), consulConfig.ServiceIdentityAuthMethod)
+		req := map[string]consul.JWTLoginRequest{}
+		identity := *service.IdentityHandle()
+		jwt, err := h.widmgr.Get(identity)
+		if err != nil {
+			h.logger.Error("error getting signed identity", "error", err)
+			mErr.Errors = append(mErr.Errors, err)
```

Comment:
```suggestion
			mErr.Errors = append(mErr.Errors, fmt.Errorf("error getting signed identity for service %s: %v", service.Name, err)
```

And similar for `prepareConsulTokensForTask` above. Some reasons for this changed:

1. having the same error context for two different operations can make debugging harder because you may not be able to tell where the `error getting signed identity` is coming from (is it a task or a service?)
2. hook errors are already logged by the alloc runner, so also logging deep here can cause the same error to be logged twice.
https://github.com/hashicorp/nomad/blob/ff928a804590611111763632388161dc711adf88/client/allocrunner/alloc_runner.go#L361-L362